src/lexer.rs

```diff
@@ -562,6 +641,97 @@ mod lexer_test {
                 ],
             },
+            TestCase {
+                input: "3",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("3", TokenKind::Integer),
+                ],
+            },
+            TestCase {
+                input: "314",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("314", TokenKind::Integer),
+                ],
+            },
+            TestCase {
+                input: "3.14",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("3.14", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "3.",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("3.", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: ".14",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    (".14", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "3e8",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("3e8", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "0.314e1",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("0.314e1", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "9.1e-31",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("9.1e-31", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "6.02e+23",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("6.02e+23", TokenKind::FloatingPoint),
+                ],
+            },
+            TestCase {
+                input: "2e",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("2e", TokenKind::Unknown),
+                ],
+            },
+            TestCase {
+                input: "2.e",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("2.e", TokenKind::Unknown),
+                ],
+            },
+            TestCase {
+                input: "2.4f",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("2.4f", TokenKind::Unknown),
+                ],
+            },
+            TestCase {
+                input: "2.4e3a",
+                skip_whitespace: false,
+                expected_tokens: vec![
+                    ("2.4e3a", TokenKind::Unknown),
+                ],
+            },
```

Comment:
Should add some test cases that combine floating-point tokens with other tokens, e.g. `let x = 3.14` and `let x = 3.2 + y` to ensure the lexer remains in a good state when it has tokens before, or tokens after, the f.p. token.

Reply:
Maybe add some tests for more complex malformed numerical types too?